business/mesh.go

```diff
@@ -133,7 +133,16 @@ func (in *MeshService) discoverKiali(ctx context.Context, clusterName string, r
 	for _, d := range services.Items {
 		kiali := convertKialiServiceToInstance(&d)
 		// If URL is already populated (because of an annotation), trust that because it's user configuration.
-		// Only guess ourselves on our own cluster.
+		// If empty, check Kiali URL configured per cluster name.
```

Comment:
IMO the kiali config should trump the annotation. They are both user configuration and, as user, it would be surprising if something I set on the kiali config was applied for any reason.